extensions/blocks/podcast-player/podcast-player.php

```diff
@@ -0,0 +1,174 @@
+
+/**
+ * Registers the block for use in Gutenberg
+ * This is done via an action so that we can disable
+ * registration if we need to.
+ */
+function register_block() {
+	jetpack_register_block(
+		BLOCK_NAME,
+		array(
+			'attributes'      => array(
+				'url'         => array(
+					'type' => 'url',
+				),
+				'itemsToShow' => array(
+					'type'    => 'integer',
+					'default' => 5,
+				),
+			),
+			'render_callback' => __NAMESPACE__ . '\render_block',
+		)
+	);
+}
+add_action( 'init', __NAMESPACE__ . '\register_block' );
+
+/**
+ * Podcast Player block registration/dependency declaration.
+ *
+ * @param array  $attributes Array containing the Podcast Player block attributes.
+ * @param string $content    String containing the Podcast Player block content.
+ *
+ * @return string
+ */
+function render_block( $attributes, $content ) {
+
+	// Test for empty URLS.
+	if ( empty( $attributes['url'] ) ) {
+		return '<p>' . esc_html__( 'No Podcast URL provided. Please enter a valid Podcast RSS feed URL.', 'jetpack' ) . '</p>';
+	}
+
+	// Test for invalid URLs.
+	if ( ! wp_http_validate_url( $attributes['url'] ) ) {
+		return '<p>' . esc_html__( 'Your podcast URL is invalid and couldn\'t be embedded. Please double check your URL.', 'jetpack' ) . '</p>';
+	}
+
+	// Sanitize the URL.
+	$attributes['url'] = esc_url_raw( $attributes['url'] );
+
+	$track_list = get_track_list( $attributes['url'], $attributes['itemsToShow'] );
+
+	if ( is_wp_error( $track_list ) ) {
+		return '<p>' . esc_html( $track_list->get_error_message() ) . '</p>';
+	}
+
+	return render_player( $track_list, $attributes );
+}
+
+/**
+ * Renders the HTML for the Podcast player and tracklist.
+ *
+ * @param array $track_list the list of podcast tracks.
+ * @param array $attributes Array containing the Podcast Player block attributes.
+ * @return string the HTML for the podcast player.
+ */
+function render_player( $track_list, $attributes ) {
+
+	$player_data = array(
+		'type'         => 'audio',
+		// Don't pass strings to JSON, will be truthy in JS.
+		'tracklist'    => true,
+		'tracknumbers' => true,
+		'images'       => true,
+		'artists'      => true,
+		'tracks'       => $track_list,
+	);
+
+	$block_classname = Jetpack_Gutenberg::block_classes( FEATURE_NAME, $attributes );
+
+	// If there are no tracks (it is possible) then display appropriate user facing error message.
+	if ( empty( $track_list ) ) {
+		return '<p>' . esc_html__( 'No tracks available to play.', 'jetpack' ) . '</p>';
+	}
+
+	ob_start();
+
+	?>
+	<div class="<?php echo esc_attr( $block_classname ); ?>">
+		<?php // Placeholder: block markup is being handled in https://github.com/Automattic/jetpack/pull/14952. ?>
+		<script type="application/json" class="wp-playlist-script"><?php echo wp_json_encode( $player_data ); ?></script>
+	</div>
+	<?php
+	/*
+	* Enqueue necessary scripts and styles.
+	*/
+	Jetpack_Gutenberg::load_assets_as_required( 'podcast-player' );
+
+	return ob_get_clean();
```

Comment:
Since we're moving towards enqueuing other JS for this to work, I would recommend getting rid of output buffering in future PRs, and instead rely on `wp_add_inline_script` if we can.

Reply:
👍 Let's try to do it [here](https://github.com/Automattic/jetpack/pull/14952).